pool.go

```diff
@@ -239,46 +253,60 @@ func (p *Pool) Start(closeFunc func()) {
 			}
 			log.Info("finish closing pool")
 
+			// wait for all on-fly retryable jobs are inserted to db
+			p.retryableWaitGroup.Wait()
+
 			// send signal to stop the program
-			p.stop <- struct{}{}
+			close(p.stop)
 			return
 		}
 	}
 }
 
 func (p *Pool) Stats() Stats {
 	return Stats{
-		PendingQueue: len(p.PrepareJobChan),
-		Queue:        len(p.JobChan),
+		RetryableQueue: len(p.RetryJobChan),
+		Queue:          len(p.JobChan),
+		RetryingJob:    atomic.LoadInt32(&p.numberOfRetryingJob),
 	}
 }
 
 func (p *Pool) PrepareRetryableJob(job JobHandler) {
+	// if pool is closed, try update job to db
+	if p.isClosed.Load().(bool) {
+		log.Info("pool closed, update retrying job to database")
+		p.updateRetryingJob(job)
+	}
 	dur := time.Until(time.Unix(job.GetNextTry(), 0))
 	if dur <= 0 {
 		return
 	}
+	atomic.AddInt32(&p.numberOfRetryingJob, 1)
```

Comment:
This `numberOfRetryingJob` seems unused

Reply:
`numberOfRetryingJob` can be used for stats